test/tools/libtesteth/ImportTest.cpp

```diff
@@ -289,13 +291,22 @@ json_spirit::mObject& ImportTest::makeAllFieldsHex(json_spirit::mObject& _o, boo
 			str = value.get_str();
 		else if (value.type() == json_spirit::array_type)
 		{
+			bool isArrayOfString = true;
 			json_spirit::mArray arr;
 			for (auto const& j: value.get_array())
 			{
+				if (j.type() != json_spirit::str_type)
+				{
+					isArrayOfString = false;
+					break;
+				}
+
 				str = j.get_str();
 				arr.push_back((str.substr(0, 2) == "0x") ? str : toCompactHexPrefixed(toInt(str), 1));
 			}
-			_o[key] = arr;
+
+			if (isArrayOfString)
+				_o[key] = arr;
```

Comment:
the function makeAllfieldsHex takes a json object 
{
  "key" : "decvalue",
  "key" : "decvalue",
  "key" : intvalue
}